docs/source/rfc-configuration-overhaul.md

```diff
@@ -0,0 +1,721 @@
+
+### Proposal 3 - Consistent Naming of Configuration Parameters
+
+#### Objective
+
+Standardise the naming conventions of Iroha's configuration parameters, enhancing their intuitiveness and consistency for improved user interaction.
+
+#### Rationale
+
+- **Clarity:** Consistent naming reduces ambiguity, allowing users to more easily understand and predict parameter names.
+- **Documentation Ease:** A unified naming convention simplifies the documentation process, ensuring coherence and easier updates.
+- **Error Minimisation:** Predictable and clear naming reduces the risk of configuration errors by users.
+
+#### Proposed Renamings
+
+Here are the proposed changes for more consistent naming:
+
+- `torii`:
+  - Rename from `p2p_addr` to `addr_p2p`.
+  - Rename from `api_url` to `addr_api`.
+  - Rename from `telemetry_url` to `addr_telemetry`.
+- `logger`:
+  - Rename from `max_log_level` to `log_level` or simply `level`.
+- `wsv`:
+  - Rename from `wasm_runtime_config` to `wasm_runtime`.
+
+For the sake of uniformity, it might be beneficial to nest the root `public_key` and `private_key` under an `iroha` namespace, resulting in `iroha.public_key` and `iroha.private_key`.
+
+Please note that this list is not exhaustive. During the configuration reference design process, we might encounter further areas of improvement and additional renaming suggestions.
+
+#### Summary
+
+This proposal aims to standardise the naming conventions within Iroha's configuration parameters, ensuring clarity and predictability, which will enhance user experience and streamline documentation efforts.
+
+#### See Also
+
+- [Proposal 1 - Use TOML](#proposal-1---use-toml)
+- [Proposal 4 - Better Aliases for ENV](#proposal-4---better-aliases-for-env)
+- [Proposal 5 - Define Deprecation and Migration Policy](#proposal-5---define-deprecation-and-migration-policy)
+- [Representation of key pair in configuration files · Issue #2135 · hyperledger/iroha](https://github.com/hyperledger/iroha/issues/2135)
+
+### Proposal 4 - Better Aliases for ENV
+
+#### Objective
+
+Introduce intuitive and standardized environment variable names, moving away from internal code-names, to simplify user experience.
+
+#### Rationale
+
+- **User Accessibility:** Generifying names improves understanding and avoids confusion. Using internal module names can be restrictive and lead to misconfiguration.
+- **Consistency:** A uniform naming convention for environment variables enhances predictability, aiding users in setting up configurations without constantly referring to documentation.
+- **Transition Support:** New aliases will coexist with existing ones, ensuring no disruptions and aiding users during the transitional phase.
+
+#### Proposed Changes
+
+1. **Rename Based on Functionality:** Shift from code-based naming to function-based naming. For example:
+   - `TORII_API_URL` becomes `API_ENDPOINT_URL`.
+   - Convert `KURA_INIT_MODE` to `BLOCK_REVALIDATION_MODE`.
```

Comment:
I feel like we should be conservative with this sort of change, lest it results in developers having to keep in mind two parallel naming schemes.

Reply:
Renaming would only have the developers remember one old scheme and a newer one.  With deprecation warnings it should be relatively easy.